src/simmate/configuration/load_settings.py

```diff
@@ -283,15 +283,43 @@ def default_settings(self) -> dict:
 
     @cached_property
     def _default_database(self) -> Path:
-        # if the user is in the (base) env or not using conda, then we will have a
-        # value of "-database.sqlite3", which why we need strip() here.
-        db_filename = (
-            self.config_directory / f"{self.conda_env}-database.sqlite3".strip("-")
-        )
-        return {
-            "engine": "django.db.backends.sqlite3",
-            "name": db_filename,
-        }
+        # !!! There may be a better way to do this, but this works for me
+        # We want to dynamically set the database backend in case the user is using
+        # a cloud database such as postgres
+        if Path(self.config_directory / f"{self.conda_env}-database.yaml").exists():
+            # if the user is in the (base) env or not using conda, then we will have a
+            # value of "-database.sqlite3", which is why we need strip() here.
+            db_filename = (
+                self.config_directory / f"{self.conda_env}-database.yaml".strip("-")
+            )
+            with open(self.config_directory / f"{self.conda_env}-database.yaml") as file:
+                yaml_file = yaml.safe_load(file)
+            default_db = {
+                "engine": yaml_file["default"]["ENGINE"],
+                "name": yaml_file["default"]["NAME"],
+                "user": yaml_file["default"]["USER"],
+                "password": yaml_file["default"]["PASSWORD"],
+                "port": yaml_file["default"]["PORT"],
+                "host": yaml_file["default"]["HOST"],
+            }
+            # try and add sslmode if it is there
+            try:
+                default_db["options"] = {"sslmode" : yaml_file["default"]["OPTIONS"]["sslmode"]}
+            except:
+                default_db["options"] = {"sslmode":"prefer"}
+            return default_db
+        
+        else:
+            # we want to use sqlite3 as our backend
+            # if the user is in the (base) env or not using conda, then we will have a
+            # value of "-database.sqlite3", which is why we need strip() here.
+            db_filename = (
+                self.config_directory / f"{self.conda_env}-database.sqlite3".strip("-")
+            )
+            return {
+                "engine": "django.db.backends.sqlite3",
+                "name": db_filename,
+            }
```

Comment:
You'll have to undo this section. There is no longer a `*-database.yaml` file, but only a `settings.yaml` file that has everything! This is updated in the docs and I also gave more details to hyesol here:
https://github.com/jacksund/simmate/discussions/683#discussioncomment-10685167

Reply:
This explains a lot :face_with_diagonal_mouth:
I was using the old `-database.yaml` file and it was of course just defaulting to sqlite3 instead. This was the first place in the code I saw that I could "fix" this so I went ahead and edited it. I've reverted this part back in the latest commit.

Unfortunately I still run into errors when running `simmate database reset`. One had to do with the psycopg2 connection to the maintenance database returning a None object, which I've fixed. Now I'm getting the following error from the `update_database` method  (full error in attachment):
```python
OperationalError: connection to server at "db-postgresql-nyc1-simmate-do-user-9931465-0.b.db.ondigitalocean.com" (206.189.189.161), port 25061 failed: FATAL:  no such database: samtest
```
When I investigate with DBeaver, the "samtest" database is shown in the database list, but it throws the same error if I try to connect to it directly. It seems like the database is successfully deleted, but something goes wrong with the creation. I've been searching for an explanation for a while but haven't found one. @jacksund any idea why this might be happening?

[error.txt](https://github.com/user-attachments/files/17466598/error.txt)

Reply:
> When I investigate with DBeaver, the "samtest" database is shown in the database list, but it throws the same error if I try to connect to it directly.

> something goes wrong with the creation

Sounds like something is going on with the database and maybe permissions. Where is your database hosted and what version of postgres is it? Digital ocean -- if so, what are you seeing on the host platform? I just checked and confirmed resetting the db works with postgres when hosted with docker locally as well as AWS RDS.

As a fallback, you can create the database manually in the DO ui and then just use `simmate database update`, rather than `reset`

Also, you can include your new migrations in this MR (the ones that add the fingerprint column to a bunch of tables). Migrations are designed to be committed to the codebase (which is something I didn't know in earlier simmate versions)

Reply:
>Sounds like something is going on with the database and maybe permissions. Where is your database hosted and what version of postgres is it? Digital ocean -- if so, what are you seeing on the host platform? I just checked and confirmed resetting the db works with postgres when hosted with docker locally as well as AWS RDS.
>
Thanks for checking! Permissions could make sense, although its odd it lets me delete full dbs if that's the case. I'm using our DigitalOcean server, but I don't have access to the ui. @scott-materials any chance you see anything on that end? If not, maybe we just try remaking the db in the DO ui and I'll run the `simmate database update` command instead.

>Also, you can include your new migrations in this MR (the ones that add the fingerprint column to a bunch of tables). Migrations are designed to be committed to the codebase (which is something I didn't know in earlier simmate versions)
>
Yeah sounds good. I was planning to once I got the postgres db going, but if I can't get it working soon I'll just go back to the sqlite3 db quick to generate them.

Reply:
> Thanks for checking! Permissions could make sense, although its odd it lets me delete full dbs if that's the case. I'm using our DigitalOcean server, but I don't have access to the ui. @scott-materials any chance you see anything on that end? If not, maybe we just try remaking the db in the DO ui and I'll run the `simmate database update` command instead.

@SWeav02 Is this still an issue?